src/NuGetTrends.Spa/src/app/shared/components/search-input/search-input.component.ts

```diff
@@ -44,13 +44,28 @@ export class SearchInputComponent implements AfterViewInit {
       distinctUntilChanged(),
       filter((value: string) => value && !!value.trim()),
       tap(() => this.isSearching = true),
-      switchMap((term: string) => this.searchNuGet(term, this.packageInteractionService.searchType)),
+      switchMap((term: string) => this.searchNuGet(term.trim(), this.packageInteractionService.searchType)),
       catchError((_, caught) => {
         this.toastr.error('Our servers are too cool (or not) to handle your request at the moment.');
         this.isSearching = false;
         return caught;
       }),
-      tap(() => this.isSearching = false));
+      tap((value) => {
+        if (value && value.length === 0) {
+          switch (this.packageInteractionService.searchType) {
+            case SearchType.NuGetPackage:
+              this.toastr.info('No packages found!');
```

Comment:
This whole method is rather long already.. maybe consider adding this in a sep func? Like `handleSearchResults`